tests/ui/collection_is_never_read.rs

```diff
@@ -0,0 +1,116 @@
+#![allow(unused)]
+#![warn(clippy::collection_is_never_read)]
+
+use std::collections::HashMap;
+
+fn main() {}
```

Comment:
I may have missed it, but I'd like to see a test using the return value of an `insert` (to some set or map) e.g. for detecting duplicates.

Something like `if my_set.insert(val).is_some() { .. }`

Reply:
Done.